felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1531,29 +1548,82 @@ func (m *bpfEndpointManager) applyProgramsToDirtyDataInterfaces() {
 			}
 
 			var parallelWG sync.WaitGroup
-			var ingressErr, xdpErr error
+			var ingressErr, xdpErr, err4, err6 error
+			var ingressAP4, egressAP4 *tc.AttachPoint
+			var ingressAP6, egressAP6 *tc.AttachPoint
+			var xdpAP4, xdpAP6 *xdp.AttachPoint
+
+			tcAttachPoint := m.calculateTCAttachPoint(iface)
+			xdpAttachPoint := &xdp.AttachPoint{
+				AttachPoint: bpf.AttachPoint{
+					Hook:     hook.XDP,
+					Iface:    iface,
+					LogLevel: m.bpfLogLevel,
+				},
+				Modes: m.xdpModes,
+			}
+
+			if m.v6 != nil {
+				parallelWG.Add(1)
+				go func() {
+					defer parallelWG.Done()
+					ingressAP6, egressAP6, xdpAP6, err6 = m.v6.applyPolicyToDataIface(iface, hepPtr, &state,
+						tcAttachPoint, xdpAttachPoint)
+				}()
+			} else if m.v4 != nil {
+				ingressAP4, egressAP4, xdpAP4, err4 = m.v4.applyPolicyToDataIface(iface, hepPtr, &state,
+					tcAttachPoint, xdpAttachPoint)
+			}
+
+			parallelWG.Wait()
```

Comment:
do you need to wait here? If you do not spin up v4 in goroutine,  it syncs and you can run the whole thin in parallel, no